### DISCUSSION TITLE
Implement new type of timer-based blocks

### DISCUSSION THREAD

**Discussion by sunshards** [proginosko/LeechBlockNG#394] [Ideas]
I recently started using an app called screenzen on my phone which works this way: for every group of apps you can select the number of times you can unlock them and the amount of time an unlock lasts. On the blocked page you can see how many unlocks you have left for the day. 
I find this very useful as most times i will not unlock to "save" my chance for later; or, if i do unlock, it's really neat that the session is on a timer, so that, even if i get distracted or sucked into infinite scrolling, after 5 or 10 minutes the time is up and i am presented with a choice again (use another unlock or go and do something productive).
I tried finding something similar on either firefox or chrome but couldn't find anything. Is there any chance this could be implemented here?

**Comment by proginosko** [comment]
LeechBlock has a feature very close to that. You can set a block on the sites you want to limit, then use the override feature to allow you access for some duration (either pre-specified or entered at the time of the override). You can also limit the number of overrides you're allowed per hour/day/week.

Would that do the job?

**Comment by sunshards** [comment]
This is actually very close to my use case. My extension wasn't fully upgraded so I couldn't see the options of limiting the number of overtimes per hour/day/week. 

The only thing is, I think this should be a per-group option and not a general option. For example, I would like to have 3 overrides per day for youtube that last 20 minutes, and 5 overrides per day for instagram that last 5 minutes, because I have different use cases for these websites. Right now, I can do this by setting them as two separate groups (that's neat), but when I activate an override, it's activated across all groups and for the same duration, so I also have to consume an override for instagram if I open youtube (and viceversa).

I also think that the user should be able to see the number of overrides they have left for that group on the page, instead of only when activating an override. (Ex: You used 3/7 overrides for group 1). This could be displayed along with the option to activate an override itself, which right now is difficult to find (only on the popup menu: i spent 10 minutes looking for it). The possibility to activate an override and display the number of total overrides on the page could be disabled or activated in the options. If implemented, this has the potential to become a whole new feature of the extension, and could be emphasized even further in the options page (but this is not as important).
Thanks for the quick response and for the work you're doing.
